Execute an error in the openj9 environment

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-fury/issues) and found no similar issues.


### Version

fury-core: 0.4.0
openjdk version "17.0.10" 2024-01-16
IBM Semeru Runtime Open Edition 17.0.10.0 (build 17.0.10+7)
Eclipse OpenJ9 VM 17.0.10.0 (build openj9-0.43.0, JRE 17 Mac OS X amd64-64-Bit Compressed References 20240116_636 (JIT enabled, AOT enabled)
OpenJ9   - 2c3d78b48
OMR      - ea8124dbc
JCL      - 2aad089841f based on jdk-17.0.10+7)


### Component(s)

Java

### Minimal reproduce step

```java

    @Getter
    @Setter
    public static class UserCacheInfoTest {

        private String extraUserId;

        private String unionId;

        private String name;

        private String jobNumber;

        private String email;

        private Long userId;

        private String mobile;

        private String avatar;

        private String position;

        private String deptName;
    }

    public static void main(String[] args) {
        UserCacheInfoTest userCacheInfoTest = new UserCacheInfoTest();
        userCacheInfoTest.setExtraUserId("1");
        userCacheInfoTest.setUnionId("1");
        userCacheInfoTest.setName("1");
        userCacheInfoTest.setJobNumber("1");
        userCacheInfoTest.setEmail("1");
        userCacheInfoTest.setUserId(0L);
        userCacheInfoTest.setMobile("1");
        userCacheInfoTest.setAvatar("1");
        userCacheInfoTest.setPosition("1");
        userCacheInfoTest.setDeptName("1");
        //ThreadLocalFury fury = new ThreadLocalFury(classLoader -> );

/
        Fury fury = Fury.builder()
                .withLanguage(Language.JAVA)
                .withCodegen(true)
                //.registerGuavaTypes(true)
                .withCompatibleMode(CompatibleMode.COMPATIBLE)
                .requireClassRegistration(false)
                .suppressClassRegistrationWarnings(true)
                .build();



        byte[] serialize = fury.serialize(userCacheInfoTest);

        UserCacheInfoTest deserialize = (UserCacheInfoTest) fury.deserialize(serialize);

        System.out.println();
    }
```




### What did you expect to see?

Normal serialization and deserialization

### What did you see instead?

```
Exception in thread "main" java.lang.RuntimeException: Create compatible serializer failed, 
class: class com.jasolar.todo.TestConfig$UserCacheInfoTest
	at io.fury.serializer.CodegenSerializer.loadCompatibleCodegenSerializer(CodegenSerializer.java:58)
	at io.fury.resolver.ClassResolver.lambda$getObjectSerializerClass$4(ClassResolver.java:976)
	at io.fury.builder.JITContext.registerSerializerJITCallback(JITContext.java:132)
	at io.fury.resolver.ClassResolver.getObjectSerializerClass(ClassResolver.java:971)
	at io.fury.resolver.ClassResolver.getSerializerClass(ClassResolver.java:894)
	at io.fury.resolver.ClassResolver.getSerializerClass(ClassResolver.java:791)
	at io.fury.resolver.ClassResolver.createSerializer(ClassResolver.java:1182)
	at io.fury.resolver.ClassResolver.getOrUpdateClassInfo(ClassResolver.java:1114)
	at io.fury.Fury.writeRef(Fury.java:342)
	at io.fury.Fury.write(Fury.java:319)
	at io.fury.Fury.serialize(Fury.java:255)
	at io.fury.Fury.serialize(Fury.java:208)
	at com.jasolar.todo.TestConfig.main(TestConfig.java:114)
Caused by: io.fury.codegen.CodegenException: Compile error: 
com.jasolar.todo.TestConfig_UserCacheInfoTestFuryCompatibleCodec_1_-2072095631_213584418:
/* 0001 */ package com.jasolar.todo;
/* 0002 */ 
/* 0003 */ import java.util.List;
/* 0004 */ import java.util.Map;
/* 0005 */ import java.util.Set;
/* 0006 */ import io.fury.Fury;
/* 0007 */ import io.fury.memory.MemoryBuffer;
/* 0008 */ import io.fury.resolver.NoRefResolver;
/* 0009 */ import io.fury.resolver.ClassInfo;
/* 0010 */ import io.fury.resolver.ClassInfoHolder;
/* 0011 */ import io.fury.resolver.ClassResolver;
/* 0012 */ import io.fury.builder.Generated;
/* 0013 */ import io.fury.serializer.CodegenSerializer.LazyInitBeanSerializer;
/* 0014 */ import io.fury.serializer.Serializers.EnumSerializer;
/* 0015 */ import io.fury.serializer.Serializer;
/* 0016 */ import io.fury.serializer.StringSerializer;
/* 0017 */ import io.fury.serializer.ObjectSerializer;
/* 0018 */ import io.fury.serializer.CompatibleSerializer;
/* 0019 */ import io.fury.serializer.collection.AbstractCollectionSerializer;
/* 0020 */ import io.fury.serializer.collection.AbstractMapSerializer;
/* 0021 */ import io.fury.builder.Generated.GeneratedCompatibleSerializer;
/* 0022 */ 
/* 0023 */ public final class TestConfig_UserCacheInfoTestFuryCompatibleCodec_1_-2072095631_213584418 extends io.fury.builder.Generated.GeneratedSerializer {
/* 0024 */ 
/* 0025 */   private final NoRefResolver refResolver;
/* 0026 */   private final ClassResolver classResolver;
/* 0027 */   private final StringSerializer strSerializer;
/* 0028 */   private final io.fury.resolver.FieldResolver fieldResolver;
/* 0029 */   private Fury fury;
/* 0030 */ 
/* 0031 */   public TestConfig_UserCacheInfoTestFuryCompatibleCodec_1_-2072095631_213584418(Fury fury, Class classType) {
/* 0032 */       super(fury, classType);
/* 0033 */       this.fury = fury;
/* 0034 */       fury.getClassResolver().setSerializerIfAbsent(classType, this);
/* 0035 */   
/* 0036 */       io.fury.resolver.RefResolver refResolver0 = fury.getRefResolver();
/* 0037 */       refResolver = ((NoRefResolver)refResolver0);
/* 0038 */       classResolver = fury.getClassResolver();
/* 0039 */       strSerializer = fury.getStringSerializer();
/* 0040 */       fieldResolver = classResolver.getFieldResolver(com.jasolar.todo.TestConfig.UserCacheInfoTest.class);
/* 0041 */   }
/* 0042 */ 
/* 0043 */   private void writeEmbedTypes4Fields(MemoryBuffer memoryBuffer, com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest1) {
/* 0044 */       memoryBuffer.writeInt(-968087463);
/* 0045 */       Object object1 = io.fury.util.Platform.getObject(userCacheInfoTest1, 16L);
/* 0046 */       String name = (String)object1;
/* 0047 */       if ((name == null)) {
/* 0048 */           memoryBuffer.writeByte(((byte)-3));
/* 0049 */       } else {
/* 0050 */           memoryBuffer.writeByte(((byte)0));
/* 0051 */           StringSerializer.writeBytesString(memoryBuffer, name);
/* 0052 */       }
/* 0053 */   }
/* 0054 */ 
/* 0055 */   private void writeEmbedTypes9Fields(MemoryBuffer memoryBuffer1, com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest2) {
/* 0056 */       memoryBuffer1.writeLong(-5248567258096544589L);
/* 0057 */       Object object00 = io.fury.util.Platform.getObject(userCacheInfoTest2, 20L);
/* 0058 */       String jobNumber = (String)object00;
/* 0059 */       if ((jobNumber == null)) {
/* 0060 */           memoryBuffer1.writeByte(((byte)-3));
/* 0061 */       } else {
/* 0062 */           memoryBuffer1.writeByte(((byte)0));
/* 0063 */           StringSerializer.writeBytesString(memoryBuffer1, jobNumber);
/* 0064 */       }
/* 0065 */       memoryBuffer1.writeLong(700233596083L);
/* 0066 */       Object object21 = io.fury.util.Platform.getObject(userCacheInfoTest2, 24L);
/* 0067 */       String email = (String)object21;
/* 0068 */       if ((email == null)) {
/* 0069 */           memoryBuffer1.writeByte(((byte)-3));
/* 0070 */       } else {
/* 0071 */           memoryBuffer1.writeByte(((byte)0));
/* 0072 */           StringSerializer.writeBytesString(memoryBuffer1, email);
/* 0073 */       }
/* 0074 */       memoryBuffer1.writeLong(40571567920307L);
/* 0075 */       Object object32 = io.fury.util.Platform.getObject(userCacheInfoTest2, 36L);
/* 0076 */       String avatar = (String)object32;
/* 0077 */       if ((avatar == null)) {
/* 0078 */           memoryBuffer1.writeByte(((byte)-3));
/* 0079 */       } else {
/* 0080 */           memoryBuffer1.writeByte(((byte)0));
/* 0081 */           StringSerializer.writeBytesString(memoryBuffer1, avatar);
/* 0082 */       }
/* 0083 */       memoryBuffer1.writeLong(53645671375027L);
/* 0084 */       Object object43 = io.fury.util.Platform.getObject(userCacheInfoTest2, 32L);
/* 0085 */       String mobile = (String)object43;
/* 0086 */       if ((mobile == null)) {
/* 0087 */           memoryBuffer1.writeByte(((byte)-3));
/* 0088 */       } else {
/* 0089 */           memoryBuffer1.writeByte(((byte)0));
/* 0090 */           StringSerializer.writeBytesString(memoryBuffer1, mobile);
/* 0091 */       }
/* 0092 */       memoryBuffer1.writeLong(62511325027491L);
/* 0093 */       Object object54 = io.fury.util.Platform.getObject(userCacheInfoTest2, 28L);
/* 0094 */       Long userId = (Long)object54;
/* 0095 */       if ((userId == null)) {
/* 0096 */           memoryBuffer1.writeByte(((byte)-3));
/* 0097 */       } else {
/* 0098 */           memoryBuffer1.writeByte(((byte)0));
/* 0099 */           memoryBuffer1.writeSliLong(userId);
/* 0100 */       }
/* 0101 */       memoryBuffer1.writeLong(3995295286467763L);
/* 0102 */       Object object65 = io.fury.util.Platform.getObject(userCacheInfoTest2, 12L);
/* 0103 */       String unionId = (String)object65;
/* 0104 */       if ((unionId == null)) {
/* 0105 */           memoryBuffer1.writeByte(((byte)-3));
/* 0106 */       } else {
/* 0107 */           memoryBuffer1.writeByte(((byte)0));
/* 0108 */           StringSerializer.writeBytesString(memoryBuffer1, unionId);
/* 0109 */       }
/* 0110 */       memoryBuffer1.writeLong(178512161548574899L);
/* 0111 */       Object object76 = io.fury.util.Platform.getObject(userCacheInfoTest2, 44L);
/* 0112 */       String deptName = (String)object76;
/* 0113 */       if ((deptName == null)) {
/* 0114 */           memoryBuffer1.writeByte(((byte)-3));
/* 0115 */       } else {
/* 0116 */           memoryBuffer1.writeByte(((byte)0));
/* 0117 */           StringSerializer.writeBytesString(memoryBuffer1, deptName);
/* 0118 */       }
/* 0119 */       memoryBuffer1.writeLong(233262162698749107L);
/* 0120 */       Object object87 = io.fury.util.Platform.getObject(userCacheInfoTest2, 40L);
/* 0121 */       String position = (String)object87;
/* 0122 */       if ((position == null)) {
/* 0123 */           memoryBuffer1.writeByte(((byte)-3));
/* 0124 */       } else {
/* 0125 */           memoryBuffer1.writeByte(((byte)0));
/* 0126 */           StringSerializer.writeBytesString(memoryBuffer1, position);
/* 0127 */       }
/* 0128 */   }
/* 0129 */ 
/* 0130 */   private void writeEmbedTypesHashFields(MemoryBuffer memoryBuffer2, com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest3) {
/* 0131 */       memoryBuffer2.writeLong(9153287876233170103L);
/* 0132 */       Object object90 = io.fury.util.Platform.getObject(userCacheInfoTest3, 8L);
/* 0133 */       String extraUserId = (String)object90;
/* 0134 */       if ((extraUserId == null)) {
/* 0135 */           memoryBuffer2.writeByte(((byte)-3));
/* 0136 */       } else {
/* 0137 */           memoryBuffer2.writeByte(((byte)0));
/* 0138 */           StringSerializer.writeBytesString(memoryBuffer2, extraUserId);
/* 0139 */       }
/* 0140 */   }
/* 0141 */ 
/* 0142 */   private long readEmbedTypes4Fields(MemoryBuffer memoryBuffer3, long value8, com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest4) {
/* 0143 */       StringSerializer strSerializer = this.strSerializer;
/* 0144 */       if ((value8 == -968087463L)) {
/* 0145 */           if ((memoryBuffer3.readByte() != ((byte)-3))) {
/* 0146 */               String string = strSerializer.readBytesString(memoryBuffer3);
/* 0147 */               io.fury.util.Platform.putObject(userCacheInfoTest4, 16L, string);
/* 0148 */           } else {
/* 0149 */               io.fury.util.Platform.putObject(userCacheInfoTest4, 16L, null);
/* 0150 */           }
/* 0151 */           value8 = memoryBuffer3.readInt();
/* 0152 */       } else {
/* 0153 */           while ((((value8 & 3L) == ((byte)1)) && (value8 < -968087463L))) {
/* 0154 */             if ((fieldResolver.skipDataBy4(memoryBuffer3, ((int)value8)) == 9223372036854775806L)) {
/* 0155 */                 return 9223372036854775806L;
/* 0156 */             }
/* 0157 */             value8 = memoryBuffer3.readInt();
/* 0158 */           }
/* 0159 */           if ((value8 == -968087463L)) {
/* 0160 */               if ((memoryBuffer3.readByte() != ((byte)-3))) {
/* 0161 */                   String string1 = strSerializer.readBytesString(memoryBuffer3);
/* 0162 */                   io.fury.util.Platform.putObject(userCacheInfoTest4, 16L, string1);
/* 0163 */               } else {
/* 0164 */                   io.fury.util.Platform.putObject(userCacheInfoTest4, 16L, null);
/* 0165 */               }
/* 0166 */               value8 = memoryBuffer3.readInt();
/* 0167 */           }
/* 0168 */       }
/* 0169 */       return value8;
/* 0170 */   }
/* 0171 */ 
/* 0172 */   private long readEmbedTypes9Fields(MemoryBuffer memoryBuffer4, long value13, com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest5) {
/* 0173 */       StringSerializer strSerializer = this.strSerializer;
/* 0174 */       io.fury.resolver.FieldResolver fieldResolver = this.fieldResolver;
/* 0175 */       if ((value13 == -5248567258096544589L)) {
/* 0176 */           if ((memoryBuffer4.readByte() != ((byte)-3))) {
/* 0177 */               String string0 = strSerializer.readBytesString(memoryBuffer4);
/* 0178 */               io.fury.util.Platform.putObject(userCacheInfoTest5, 20L, string0);
/* 0179 */           } else {
/* 0180 */               io.fury.util.Platform.putObject(userCacheInfoTest5, 20L, null);
/* 0181 */           }
/* 0182 */           value13 = memoryBuffer4.readLong();
/* 0183 */       } else {
/* 0184 */           while ((((value13 & 7L) == ((byte)3)) && (value13 < -5248567258096544589L))) {
/* 0185 */             if ((fieldResolver.skipDataBy8(memoryBuffer4, value13) == 9223372036854775806L)) {
/* 0186 */                 return 9223372036854775806L;
/* 0187 */             }
/* 0188 */             value13 = memoryBuffer4.readLong();
/* 0189 */           }
/* 0190 */           if ((value13 == -5248567258096544589L)) {
/* 0191 */               if ((memoryBuffer4.readByte() != ((byte)-3))) {
/* 0192 */                   String string2 = strSerializer.readBytesString(memoryBuffer4);
/* 0193 */                   io.fury.util.Platform.putObject(userCacheInfoTest5, 20L, string2);
/* 0194 */               } else {
/* 0195 */                   io.fury.util.Platform.putObject(userCacheInfoTest5, 20L, null);
/* 0196 */               }
/* 0197 */               value13 = memoryBuffer4.readLong();
/* 0198 */           }
/* 0199 */       }
/* 0200 */       if ((value13 == 700233596083L)) {
/* 0201 */           if ((memoryBuffer4.readByte() != ((byte)-3))) {
/* 0202 */               String string3 = strSerializer.readBytesString(memoryBuffer4);
/* 0203 */               io.fury.util.Platform.putObject(userCacheInfoTest5, 24L, string3);
/* 0204 */           } else {
/* 0205 */               io.fury.util.Platform.putObject(userCacheInfoTest5, 24L, null);
/* 0206 */           }
/* 0207 */           value13 = memoryBuffer4.readLong();
/* 0208 */       } else {
/* 0209 */           while ((((value13 & 7L) == ((byte)3)) && (value13 < 700233596083L))) {
/* 0210 */             if ((fieldResolver.skipDataBy8(memoryBuffer4, value13) == 9223372036854775806L)) {
/* 0211 */                 return 9223372036854775806L;
/* 0212 */             }
/* 0213 */             value13 = memoryBuffer4.readLong();
/* 0214 */           }
/* 0215 */           if ((value13 == 700233596083L)) {
/* 0216 */               if ((memoryBuffer4.readByte() != ((byte)-3))) {
/* 0217 */                   String string4 = strSerializer.readBytesString(memoryBuffer4);
/* 0218 */                   io.fury.util.Platform.putObject(userCacheInfoTest5, 24L, string4);
/* 0219 */               } else {
/* 0220 */                   io.fury.util.Platform.putObject(userCacheInfoTest5, 24L, null);
/* 0221 */               }
/* 0222 */               value13 = memoryBuffer4.readLong();
/* 0223 */           }
/* 0224 */       }
/* 0225 */       if ((value13 == 40571567920307L)) {
/* 0226 */           if ((memoryBuffer4.readByte() != ((byte)-3))) {
/* 0227 */               String string5 = strSerializer.readBytesString(memoryBuffer4);
/* 0228 */               io.fury.util.Platform.putObject(userCacheInfoTest5, 36L, string5);
/* 0229 */           } else {
/* 0230 */               io.fury.util.Platform.putObject(userCacheInfoTest5, 36L, null);
/* 0231 */           }
/* 0232 */           value13 = memoryBuffer4.readLong();
/* 0233 */       } else {
/* 0234 */           while ((((value13 & 7L) == ((byte)3)) && (value13 < 40571567920307L))) {
/* 0235 */             if ((fieldResolver.skipDataBy8(memoryBuffer4, value13) == 9223372036854775806L)) {
/* 0236 */                 return 9223372036854775806L;
/* 0237 */             }
/* 0238 */             value13 = memoryBuffer4.readLong();
/* 0239 */           }
/* 0240 */           if ((value13 == 40571567920307L)) {
/* 0241 */               if ((memoryBuffer4.readByte() != ((byte)-3))) {
/* 0242 */                   String string6 = strSerializer.readBytesString(memoryBuffer4);
/* 0243 */                   io.fury.util.Platform.putObject(userCacheInfoTest5, 36L, string6);
/* 0244 */               } else {
/* 0245 */                   io.fury.util.Platform.putObject(userCacheInfoTest5, 36L, null);
/* 0246 */               }
/* 0247 */               value13 = memoryBuffer4.readLong();
/* 0248 */           }
/* 0249 */       }
/* 0250 */       return value13;
/* 0251 */   }
/* 0252 */ 
/* 0253 */   private long readEmbedTypes9Fields1(MemoryBuffer memoryBuffer5, long value26, com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest6) {
/* 0254 */       StringSerializer strSerializer = this.strSerializer;
/* 0255 */       io.fury.resolver.FieldResolver fieldResolver = this.fieldResolver;
/* 0256 */       if ((value26 == 53645671375027L)) {
/* 0257 */           if ((memoryBuffer5.readByte() != ((byte)-3))) {
/* 0258 */               String string7 = strSerializer.readBytesString(memoryBuffer5);
/* 0259 */               io.fury.util.Platform.putObject(userCacheInfoTest6, 32L, string7);
/* 0260 */           } else {
/* 0261 */               io.fury.util.Platform.putObject(userCacheInfoTest6, 32L, null);
/* 0262 */           }
/* 0263 */           value26 = memoryBuffer5.readLong();
/* 0264 */       } else {
/* 0265 */           while ((((value26 & 7L) == ((byte)3)) && (value26 < 53645671375027L))) {
/* 0266 */             if ((fieldResolver.skipDataBy8(memoryBuffer5, value26) == 9223372036854775806L)) {
/* 0267 */                 return 9223372036854775806L;
/* 0268 */             }
/* 0269 */             value26 = memoryBuffer5.readLong();
/* 0270 */           }
/* 0271 */           if ((value26 == 53645671375027L)) {
/* 0272 */               if ((memoryBuffer5.readByte() != ((byte)-3))) {
/* 0273 */                   String string8 = strSerializer.readBytesString(memoryBuffer5);
/* 0274 */                   io.fury.util.Platform.putObject(userCacheInfoTest6, 32L, string8);
/* 0275 */               } else {
/* 0276 */                   io.fury.util.Platform.putObject(userCacheInfoTest6, 32L, null);
/* 0277 */               }
/* 0278 */               value26 = memoryBuffer5.readLong();
/* 0279 */           }
/* 0280 */       }
/* 0281 */       if ((value26 == 62511325027491L)) {
/* 0282 */           if ((memoryBuffer5.readByte() != ((byte)-3))) {
/* 0283 */               long value31 = memoryBuffer5.readSliLong();
/* 0284 */               io.fury.util.Platform.putObject(userCacheInfoTest6, 28L, ((Long)value31));
/* 0285 */           } else {
/* 0286 */               io.fury.util.Platform.putObject(userCacheInfoTest6, 28L, null);
/* 0287 */           }
/* 0288 */           value26 = memoryBuffer5.readLong();
/* 0289 */       } else {
/* 0290 */           while ((((value26 & 7L) == ((byte)3)) && (value26 < 62511325027491L))) {
/* 0291 */             if ((fieldResolver.skipDataBy8(memoryBuffer5, value26) == 9223372036854775806L)) {
/* 0292 */                 return 9223372036854775806L;
/* 0293 */             }
/* 0294 */             value26 = memoryBuffer5.readLong();
/* 0295 */           }
/* 0296 */           if ((value26 == 62511325027491L)) {
/* 0297 */               if ((memoryBuffer5.readByte() != ((byte)-3))) {
/* 0298 */                   long value34 = memoryBuffer5.readSliLong();
/* 0299 */                   io.fury.util.Platform.putObject(userCacheInfoTest6, 28L, ((Long)value34));
/* 0300 */               } else {
/* 0301 */                   io.fury.util.Platform.putObject(userCacheInfoTest6, 28L, null);
/* 0302 */               }
/* 0303 */               value26 = memoryBuffer5.readLong();
/* 0304 */           }
/* 0305 */       }
/* 0306 */       if ((value26 == 3995295286467763L)) {
/* 0307 */           if ((memoryBuffer5.readByte() != ((byte)-3))) {
/* 0308 */               String string9 = strSerializer.readBytesString(memoryBuffer5);
/* 0309 */               io.fury.util.Platform.putObject(userCacheInfoTest6, 12L, string9);
/* 0310 */           } else {
/* 0311 */               io.fury.util.Platform.putObject(userCacheInfoTest6, 12L, null);
/* 0312 */           }
/* 0313 */           value26 = memoryBuffer5.readLong();
/* 0314 */       } else {
/* 0315 */           while ((((value26 & 7L) == ((byte)3)) && (value26 < 3995295286467763L))) {
/* 0316 */             if ((fieldResolver.skipDataBy8(memoryBuffer5, value26) == 9223372036854775806L)) {
/* 0317 */                 return 9223372036854775806L;
/* 0318 */             }
/* 0319 */             value26 = memoryBuffer5.readLong();
/* 0320 */           }
/* 0321 */           if ((value26 == 3995295286467763L)) {
/* 0322 */               if ((memoryBuffer5.readByte() != ((byte)-3))) {
/* 0323 */                   String string10 = strSerializer.readBytesString(memoryBuffer5);
/* 0324 */                   io.fury.util.Platform.putObject(userCacheInfoTest6, 12L, string10);
/* 0325 */               } else {
/* 0326 */                   io.fury.util.Platform.putObject(userCacheInfoTest6, 12L, null);
/* 0327 */               }
/* 0328 */               value26 = memoryBuffer5.readLong();
/* 0329 */           }
/* 0330 */       }
/* 0331 */       return value26;
/* 0332 */   }
/* 0333 */ 
/* 0334 */   private long readEmbedTypes9Fields2(MemoryBuffer memoryBuffer6, long value41, com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest7) {
/* 0335 */       StringSerializer strSerializer = this.strSerializer;
/* 0336 */       io.fury.resolver.FieldResolver fieldResolver = this.fieldResolver;
/* 0337 */       if ((value41 == 178512161548574899L)) {
/* 0338 */           if ((memoryBuffer6.readByte() != ((byte)-3))) {
/* 0339 */               String string11 = strSerializer.readBytesString(memoryBuffer6);
/* 0340 */               io.fury.util.Platform.putObject(userCacheInfoTest7, 44L, string11);
/* 0341 */           } else {
/* 0342 */               io.fury.util.Platform.putObject(userCacheInfoTest7, 44L, null);
/* 0343 */           }
/* 0344 */           value41 = memoryBuffer6.readLong();
/* 0345 */       } else {
/* 0346 */           while ((((value41 & 7L) == ((byte)3)) && (value41 < 178512161548574899L))) {
/* 0347 */             if ((fieldResolver.skipDataBy8(memoryBuffer6, value41) == 9223372036854775806L)) {
/* 0348 */                 return 9223372036854775806L;
/* 0349 */             }
/* 0350 */             value41 = memoryBuffer6.readLong();
/* 0351 */           }
/* 0352 */           if ((value41 == 178512161548574899L)) {
/* 0353 */               if ((memoryBuffer6.readByte() != ((byte)-3))) {
/* 0354 */                   String string12 = strSerializer.readBytesString(memoryBuffer6);
/* 0355 */                   io.fury.util.Platform.putObject(userCacheInfoTest7, 44L, string12);
/* 0356 */               } else {
/* 0357 */                   io.fury.util.Platform.putObject(userCacheInfoTest7, 44L, null);
/* 0358 */               }
/* 0359 */               value41 = memoryBuffer6.readLong();
/* 0360 */           }
/* 0361 */       }
/* 0362 */       if ((value41 == 233262162698749107L)) {
/* 0363 */           if ((memoryBuffer6.readByte() != ((byte)-3))) {
/* 0364 */               String string13 = strSerializer.readBytesString(memoryBuffer6);
/* 0365 */               io.fury.util.Platform.putObject(userCacheInfoTest7, 40L, string13);
/* 0366 */           } else {
/* 0367 */               io.fury.util.Platform.putObject(userCacheInfoTest7, 40L, null);
/* 0368 */           }
/* 0369 */           value41 = memoryBuffer6.readLong();
/* 0370 */       } else {
/* 0371 */           while ((((value41 & 7L) == ((byte)3)) && (value41 < 233262162698749107L))) {
/* 0372 */             if ((fieldResolver.skipDataBy8(memoryBuffer6, value41) == 9223372036854775806L)) {
/* 0373 */                 return 9223372036854775806L;
/* 0374 */             }
/* 0375 */             value41 = memoryBuffer6.readLong();
/* 0376 */           }
/* 0377 */           if ((value41 == 233262162698749107L)) {
/* 0378 */               if ((memoryBuffer6.readByte() != ((byte)-3))) {
/* 0379 */                   String string14 = strSerializer.readBytesString(memoryBuffer6);
/* 0380 */                   io.fury.util.Platform.putObject(userCacheInfoTest7, 40L, string14);
/* 0381 */               } else {
/* 0382 */                   io.fury.util.Platform.putObject(userCacheInfoTest7, 40L, null);
/* 0383 */               }
/* 0384 */               value41 = memoryBuffer6.readLong();
/* 0385 */           }
/* 0386 */       }
/* 0387 */       return value41;
/* 0388 */   }
/* 0389 */ 
/* 0390 */   private long readEmbedTypesHashFields(MemoryBuffer memoryBuffer7, long value50, com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest8) {
/* 0391 */       StringSerializer strSerializer = this.strSerializer;
/* 0392 */       if ((value50 == 9153287876233170103L)) {
/* 0393 */           if ((memoryBuffer7.readByte() != ((byte)-3))) {
/* 0394 */               String string15 = strSerializer.readBytesString(memoryBuffer7);
/* 0395 */               io.fury.util.Platform.putObject(userCacheInfoTest8, 8L, string15);
/* 0396 */           } else {
/* 0397 */               io.fury.util.Platform.putObject(userCacheInfoTest8, 8L, null);
/* 0398 */           }
/* 0399 */           value50 = memoryBuffer7.readLong();
/* 0400 */       } else {
/* 0401 */           while ((((value50 & 7L) == ((byte)7)) && (value50 < 9153287876233170103L))) {
/* 0402 */             if ((fieldResolver.skipDataBy8(memoryBuffer7, value50) == 9223372036854775806L)) {
/* 0403 */                 return 9223372036854775806L;
/* 0404 */             }
/* 0405 */             value50 = memoryBuffer7.readLong();
/* 0406 */           }
/* 0407 */           if ((value50 == 9153287876233170103L)) {
/* 0408 */               if ((memoryBuffer7.readByte() != ((byte)-3))) {
/* 0409 */                   String string16 = strSerializer.readBytesString(memoryBuffer7);
/* 0410 */                   io.fury.util.Platform.putObject(userCacheInfoTest8, 8L, string16);
/* 0411 */               } else {
/* 0412 */                   io.fury.util.Platform.putObject(userCacheInfoTest8, 8L, null);
/* 0413 */               }
/* 0414 */               value50 = memoryBuffer7.readLong();
/* 0415 */           }
/* 0416 */       }
/* 0417 */       return value50;
/* 0418 */   }
/* 0419 */ 
/* 0420 */   @Override public final void write(MemoryBuffer buffer, Object obj) {
/* 0421 */       com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest9 = (com.jasolar.todo.TestConfig.UserCacheInfoTest)obj;
/* 0422 */       this.writeEmbedTypes4Fields(buffer, userCacheInfoTest9);
/* 0423 */       this.writeEmbedTypes9Fields(buffer, userCacheInfoTest9);
/* 0424 */       this.writeEmbedTypesHashFields(buffer, userCacheInfoTest9);
/* 0425 */       buffer.writeLong(9223372036854775806L);
/* 0426 */   }
/* 0427 */ 
/* 0428 */   @Override public final Object read(MemoryBuffer buffer) {
/* 0429 */       io.fury.resolver.FieldResolver fieldResolver = this.fieldResolver;
/* 0430 */       com.jasolar.todo.TestConfig.UserCacheInfoTest userCacheInfoTest10 = new com.jasolar.todo.TestConfig.UserCacheInfoTest();
/* 0431 */       refResolver.reference(userCacheInfoTest10);
/* 0432 */       long partFieldInfo = buffer.readInt();
/* 0433 */       while ((((partFieldInfo & 3L) == ((byte)1)) && (partFieldInfo < -968087463L))) {
/* 0434 */         if ((fieldResolver.skipDataBy4(buffer, ((int)partFieldInfo)) == 9223372036854775806L)) {
/* 0435 */             return userCacheInfoTest10;
/* 0436 */         }
/* 0437 */         partFieldInfo = buffer.readInt();
/* 0438 */       }
/* 0439 */       partFieldInfo = this.readEmbedTypes4Fields(buffer, partFieldInfo, userCacheInfoTest10);
/* 0440 */       if ((partFieldInfo == 9223372036854775806L)) {
/* 0441 */           return userCacheInfoTest10;
/* 0442 */       }
/* 0443 */       while (((partFieldInfo & 3L) == ((byte)1))) {
/* 0444 */         if ((fieldResolver.skipDataBy4(buffer, ((int)partFieldInfo)) == 9223372036854775806L)) {
/* 0445 */             return userCacheInfoTest10;
/* 0446 */         }
/* 0447 */         partFieldInfo = buffer.readInt();
/* 0448 */       }
/* 0449 */       long value55 = buffer.readInt();
/* 0450 */       partFieldInfo = ((value55 << 32) | (partFieldInfo & 4294967295L));
/* 0451 */       while ((((partFieldInfo & 7L) == ((byte)3)) && (partFieldInfo < -5248567258096544589L))) {
/* 0452 */         if ((fieldResolver.skipDataBy8(buffer, partFieldInfo) == 9223372036854775806L)) {
/* 0453 */             return userCacheInfoTest10;
/* 0454 */         }
/* 0455 */         partFieldInfo = buffer.readLong();
/* 0456 */       }
/* 0457 */       partFieldInfo = this.readEmbedTypes9Fields(buffer, partFieldInfo, userCacheInfoTest10);
/* 0458 */       if ((partFieldInfo == 9223372036854775806L)) {
/* 0459 */           return userCacheInfoTest10;
/* 0460 */       }
/* 0461 */       partFieldInfo = this.readEmbedTypes9Fields1(buffer, partFieldInfo, userCacheInfoTest10);
/* 0462 */       if ((partFieldInfo == 9223372036854775806L)) {
/* 0463 */           return userCacheInfoTest10;
/* 0464 */       }
/* 0465 */       partFieldInfo = this.readEmbedTypes9Fields2(buffer, partFieldInfo, userCacheInfoTest10);
/* 0466 */       if ((partFieldInfo == 9223372036854775806L)) {
/* 0467 */           return userCacheInfoTest10;
/* 0468 */       }
/* 0469 */       while (((partFieldInfo & 7L) == ((byte)3))) {
/* 0470 */         if ((fieldResolver.skipDataBy8(buffer, partFieldInfo) == 9223372036854775806L)) {
/* 0471 */             return userCacheInfoTest10;
/* 0472 */         }
/* 0473 */         partFieldInfo = buffer.readLong();
/* 0474 */       }
/* 0475 */       while ((((partFieldInfo & 7L) == ((byte)7)) && (partFieldInfo < 9153287876233170103L))) {
/* 0476 */         if ((fieldResolver.skipDataBy8(buffer, partFieldInfo) == 9223372036854775806L)) {
/* 0477 */             return userCacheInfoTest10;
/* 0478 */         }
/* 0479 */         partFieldInfo = buffer.readLong();
/* 0480 */       }
/* 0481 */       partFieldInfo = this.readEmbedTypesHashFields(buffer, partFieldInfo, userCacheInfoTest10);
/* 0482 */       if ((partFieldInfo == 9223372036854775806L)) {
/* 0483 */           return userCacheInfoTest10;
/* 0484 */       }
/* 0485 */       while (((partFieldInfo & 7L) == ((byte)7))) {
/* 0486 */         if ((fieldResolver.skipDataBy8(buffer, partFieldInfo) == 9223372036854775806L)) {
/* 0487 */             return userCacheInfoTest10;
/* 0488 */         }
/* 0489 */         partFieldInfo = buffer.readLong();
/* 0490 */       }
/* 0491 */       fieldResolver.skipEndFields(buffer, partFieldInfo);
/* 0492 */       return userCacheInfoTest10;
/* 0493 */   }
/* 0494 */ 
/* 0495 */ }
	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:126)
	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:64)
	at io.fury.codegen.CodeGenerator.compile(CodeGenerator.java:144)
	at io.fury.builder.CodecUtils.loadOrGenCodecClass(CodecUtils.java:91)
	at io.fury.builder.CodecUtils.loadOrGenCompatibleCodecClass(CodecUtils.java:64)
	at io.fury.builder.CodecUtils.loadOrGenCompatibleCodecClass(CodecUtils.java:56)
	at io.fury.serializer.CodegenSerializer.loadCompatibleCodegenSerializer(CodegenSerializer.java:55)
	... 12 more
Caused by: io.fury.shaded.org.codehaus.commons.compiler.CompileException: File 'com/jasolar/todo/TestConfig_UserCacheInfoTestFuryCompatibleCodec_1_-2072095631_213584418.java', Line 23, Column 70: '{' expected instead of '-'
	at io.fury.shaded.org.codehaus.janino.TokenStreamImpl.compileException(TokenStreamImpl.java:362)
	at io.fury.shaded.org.codehaus.janino.TokenStreamImpl.read(TokenStreamImpl.java:149)
	at io.fury.shaded.org.codehaus.janino.Parser.read(Parser.java:3802)
	at io.fury.shaded.org.codehaus.janino.Parser.parseClassBody(Parser.java:851)
	at io.fury.shaded.org.codehaus.janino.Parser.parseClassDeclarationRest(Parser.java:746)
	at io.fury.shaded.org.codehaus.janino.Parser.parsePackageMemberTypeDeclarationRest(Parser.java:492)
	at io.fury.shaded.org.codehaus.janino.Parser.parseAbstractCompilationUnit(Parser.java:267)
	at io.fury.shaded.org.codehaus.janino.Compiler.parseAbstractCompilationUnit(Compiler.java:316)
	at io.fury.shaded.org.codehaus.janino.Compiler.compile2(Compiler.java:236)
	at io.fury.shaded.org.codehaus.janino.Compiler.compile(Compiler.java:213)
	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:106)

```

### Anything Else?

Run the same code, use the openj9-jdk17 error, use the zuul-jdk17 is indeed normal.I suspect openj9 does not support fury, but I don't know how to change it.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/java/fury-core/src/main/java/org/apache/fury/codegen/CodeGenerator.java
+++ b/java/fury-core/src/main/java/org/apache/fury/codegen/CodeGenerator.java
@@ -344,10 +344,12 @@ public class CodeGenerator {
     }
     // classLoader will be null for jdk classes.
     ClassLoader classLoader = cls.getClassLoader();
+    // Hashcode may be negative in open-j9 jdk. While using `abs` to remove sign works fine too.
+    // it's still possible that hashCodes of two objects only differ in sign.
     if (classLoader == null) {
-      return String.valueOf(cls.hashCode());
+      return String.valueOf(cls.hashCode()).replace("-", "_");
     } else {
-      return String.format("%s_%s", classLoader.hashCode(), cls.hashCode());
+      return String.format("%s_%s", classLoader.hashCode(), cls.hashCode()).replace("-", "_");
     }
   }
 

